pkg/operator/staticpod/installerpod/copy.go

```diff
@@ -16,54 +16,52 @@ import (
 // It will retry on API server connection errors except not found error which is fatal for non-optional secrets.
 // In case the secret is optional and we fail to get it, no error is returned and the secret returning is nil.
 func (o *InstallOptions) getSecretWithRetry(ctx context.Context, secretNamePrefix string, isOptional bool) (*v1.Secret, error) {
-	var secret *v1.Secret
-	retryErr := wait.PollImmediateUntil(200*time.Millisecond, func() (done bool, err error) {
-		secret, err = o.KubeClient.CoreV1().Secrets(o.Namespace).Get(o.nameFor(secretNamePrefix), metav1.GetOptions{})
+	var resultSecret *v1.Secret = nil
+	retryErr := wait.PollImmediateUntil(200*time.Millisecond, func() (bool, error) {
+		secret, err := o.KubeClient.CoreV1().Secrets(o.Namespace).Get(o.nameFor(secretNamePrefix), metav1.GetOptions{})
 		switch {
 		case errors.IsNotFound(err):
-			// if secret is optional, report success even when the secret was not found
 			if isOptional {
 				err = nil
 			}
 			return true, err
 		case err != nil:
-			if !isOptional {
-				glog.Warningf("Failed to get secret %q: %v (will retry)", o.Namespace+"/"+o.nameFor(secretNamePrefix), err)
-			}
-			// if secret is optional, report success on any error and never retry
-			return isOptional, nil
+			glog.Warningf("Failed to get secret %s/%s: %v (will retry)", o.Namespace, o.nameFor(secretNamePrefix), err)
+			return false, nil
 		default:
+			resultSecret = secret
 			return true, nil
 		}
 	}, ctx.Done())
-
-	return secret, retryErr
+	if retryErr != nil {
```

Comment:
if clause is not needed, is it?

Reply:
no, but i'm in paranoid mode now.

Reply:
it's logic, nothing to be paranoid about ;-)

Reply:
just wanted to make sure I never return secret when there is an error.

Reply:
ok, removed this.